File: frontend/src/types/harness.ts
export type ValidationGateType = 'multiple_choice' | 'true_false' | 'mark_complete'
export type GatingModel = 'strict' | 'advisory' | 'open_world'
export type ScaffoldingLevel = 'full_scaffold' | 'guided' | 'prompted' | 'raw_mode'
export type ChunkingDensity = 'micro_atomic' | 'compact' | 'standard' | 'academic'
export type PacingMode = 'sprint' | 'steady' | 'flexible' | 'immersion'
export type TutorTone = 'formal' | 'direct' | 'conversational' | 'energetic'
export type TutorPace = 'slow' | 'steady' | 'fast'
export type TutorRegister = 'fundamentals' | 'practitioner' | 'expert_peer'
export type FeedbackStyle = 'direct' | 'explanatory' | 'socratic' | 'encouraging'
export type AssessmentPreference = 'drill_heavy' | 'project_anchored' | 'mixed' | 'exam_simulated'
export type LearningPhilosophy = 'direct_instruction' | 'socratic' | 'constructivist' | 'spaced_practice'

export interface ValidationGate {
  type: ValidationGateType
  pass_threshold: number
  question_count: number
}

export interface CoreLesson {
  lesson_id: string
  lesson_title: string
  order: number
  prerequisites: string[]
  validation: ValidationGate
}

export interface CoreModule {
  module_id: string
  module_title: string
  order: number
  lessons: CoreLesson[]
}

export interface CoreLayer {
  course_title: string
  terminal_objective: string
  estimated_weeks: number
  modules: CoreModule[]
}

export interface MediaMix {
  video: number
  interactive: number
  text: number
  audio?: number
  visual?: number
}

export interface TutorPersona {
  pace: TutorPace
  tone: TutorTone
  register: TutorRegister
}

export interface TrackLayer {
  path: string
  scaffolding: ScaffoldingLevel
  gating: GatingModel
  chunking: ChunkingDensity
  media_mix: MediaMix
  pacing_mode: PacingMode
  feedback_style: FeedbackStyle
  assessment_preference: AssessmentPreference
  tutor: TutorPersona
  session_cap_minutes: number
  srs_enabled: boolean
}

export interface Harness {
  core: CoreLayer
  track: TrackLayer
}

File: backend/app/models/harness.py
from pydantic import BaseModel
from typing import Optional, Literal

class ValidationGate(BaseModel):
    type: Literal["multiple_choice", "true_false", "mark_complete"]
    pass_threshold: float = 0.8
    question_count: int = 5

class CoreLesson(BaseModel):
    lesson_id: str
    lesson_title: str
    order: int
    prerequisites: list[str] = []
    validation: ValidationGate

class CoreModule(BaseModel):
    module_id: str
    module_title: str
    order: int
    lessons: list[CoreLesson]

class CoreLayer(BaseModel):
    course_title: str
    terminal_objective: str
    estimated_weeks: int
    modules: list[CoreModule]

class MediaMix(BaseModel):
    video: int = 40
    interactive: int = 40
    text: int = 20

class TutorPersona(BaseModel):
    pace: str = "steady"
    tone: str = "direct"
    register: str = "practitioner"

class TrackLayer(BaseModel):
    path: str
    scaffolding: str = "guided"
    gating: str = "strict"
    chunking: str = "standard"
    media_mix: MediaMix = MediaMix()
    pacing_mode: str = "steady"
    feedback_style: str = "direct"
    assessment_preference: str = "mixed"
    tutor: TutorPersona = TutorPersona()
    session_cap_minutes: int = 60
    srs_enabled: bool = True

class Harness(BaseModel):
    core: CoreLayer
    track: TrackLayer
